snowshu/adapters/target_adapters/base_target_adapter.py

```diff
@@ -178,19 +180,22 @@ def initialize_replica(self,
     def _init_image(self,
                     source_adapter_name: str) -> None:
         shdocker = SnowShuDocker()
-
-        image_architecture = shdocker.get_docker_image_attributes(self.DOCKER_IMAGE) \
-            .get('Architecture', 'Unknown architecture')
-
-        if image_architecture.lower() == LOCAL_ARCHITECTURE.lower():
-            logger.info(f'The architecture of the docker image[{self.DOCKER_IMAGE}] is {image_architecture.lower()}')
-        else:
-            logger.warning(f"""There is a mismatch between local and docker image architectures: '
-                                   The local architecture is {LOCAL_ARCHITECTURE} 
-                                   The image architecture is [{self.DOCKER_IMAGE}]({image_architecture})""")
+        try:
+            image_architecture = shdocker.get_docker_image_attributes(self.DOCKER_IMAGE) \
+                .get('Architecture', 'Unknown architecture')
+
+            if image_architecture.lower() == LOCAL_ARCHITECTURE.lower():
+                logger.info(
+                    f'The architecture of the docker image[{self.DOCKER_IMAGE}] is {image_architecture.lower()}')
+            else:
+                logger.warning(f"""There is a mismatch between local and docker image architectures: '
+                                    The local architecture is {LOCAL_ARCHITECTURE} 
+                                    The image architecture is [{self.DOCKER_IMAGE}]({image_architecture})""")
+        except docker.errors.ImageNotFound:
+            logger.info('No existing target image found, creating')
```

Comment:
Yeah, before I changed it it would crash if you don't have any postgres image locally

Reply:
Actually, I think this entire block can be removed since I removed the image.get(image_name) call further down the line. If the images are cached, the pull command will be nearly instant and will guarantee we pull the image of correct arch, instead of whatever latest postgres:12 we have